tests/test_process_node.py

```diff
@@ -78,189 +94,100 @@ class TestProcessQuery(unittest.TestCase):
 
         assert not queried_proc.get_asset()
 
     @hypothesis.settings(deadline=None)
     @given(
-        asset_node_key=st.uuids(),
-        hostname=st.text(),
-        process_node_key=st.uuids(),
-        process_id=st.integers(min_value=1, max_value=2 ** 32),
-        created_timestamp=st.integers(min_value=0, max_value=2 ** 48),
-        terminate_time=st.integers(min_value=0, max_value=2 ** 48),
-        image_name=st.text(),
-        process_name=st.text(),
-        arguments=st.text(),
+        asset_props=asset_props(), process_props=process_props(),
     )
     def test_single_process_connected_to_asset_node(
-        self,
-        asset_node_key,
-        hostname,
-        process_node_key,
-        process_id,
-        created_timestamp,
-        terminate_time,
-        image_name,
-        process_name,
-        arguments,
+        self, asset_props: AssetProps, process_props: ProcessProps,
     ):
-        asset_node_key = node_key_for_test(self, asset_node_key)
-        process_node_key = node_key_for_test(self, process_node_key)
         local_client = DgraphClient(DgraphClientStub("localhost:9080"))
 
-        created_asset_node = get_or_create_asset_node(
-            local_client, asset_node_key, hostname=hostname
-        )
-
-        created_process_node = get_or_create_process_node(
-            local_client,
-            process_node_key,
-            process_id,
-            arguments,
-            created_timestamp,
-            terminate_time,
-            image_name,
-            process_name,
-        )
+        created_asset = get_or_create_asset(self, local_client, asset_props)
+        created_proc = get_or_create_process(self, local_client, process_props)
 
         create_edge(
-            local_client,
-            created_asset_node.uid,
-            "asset_processes",
-            created_process_node.uid,
+            local_client, created_asset.uid, "asset_processes", created_proc.uid,
         )
 
         # Setup complete, do some queries
 
         queried_proc = (
             ProcessQuery()
-            .with_asset(AssetQuery().with_hostname(hostname))
-            .query_first(local_client, contains_node_key=process_node_key)
+            .with_asset(AssetQuery().with_hostname(created_asset.get_hostname()))
+            .query_first(local_client, contains_node_key=created_proc.node_key)
         )
-
-        assert process_id == queried_proc.get_process_id()
-        assert process_node_key == queried_proc.node_key
-        assert "Process" == queried_proc.get_node_type()
-        assert process_id == queried_proc.get_process_id()
-        assert arguments == queried_proc.get_arguments()
-        assert created_timestamp == queried_proc.get_created_timestamp()
-        assert terminate_time == queried_proc.get_terminate_time()
-        assert image_name == queried_proc.get_image_name()
-        assert process_name == queried_proc.get_process_name()
-
+        assert queried_proc
```

Comment:
this test is slightly different - instead of relying on .get_PROP() to retrieve, I'm fetching all props and dynamically comparing the properties dicts.